### PR TITLE
Abstract db access from Postfix and Dovecot

### PR DESCRIPTION
**Pull request by kaiyou**
This is a work in progress but I guess the review can start already as 90% of it is testable.

Most of the original work is discussed in #420 and #445. The general idea is that  

- Podop (https://github.com/mailu/podop) is a proxy framework between Postfix/Dovecot and any Python available service, including notably a REST API
- A fairly simple (and still a bit too tightly dependent on Postfix/Dovecot structures) API was written as part of the internal REST API already leveraged for authentication, the API includes transparent authentication for Dovecot, domain, alias, mailbox and transport resolution for Postfix, quotas, sieve scripts.
- A simple sieve script as a Jinja template was written, removing the need for extdata in sieve.
- Dovecot and Postfix were bound to the admin container API using Podop, embedded as a side process in both start scripts.
- Fetchmail was bound to a simple internal API on the admin container.

No change should be required in a standard compose environment. There are still a lot of tests to run before declaring this equivalent to the current database queries.

### CODE REVIEW

**Comment by kaiyou** [comment]
I just pushed the last changes required for isofunctionality with the current setup, so reviews are definitely welcome :)

**Comment by muhlemmer** [comment]
Cool. After the weekend I'll install a server with this branch :1st_place_medal:

**Comment by kaiyou** [comment]
1. I am working on fixing incoming and outgoing email on this branch
2. -
3. Redis was used for storing current quota and was replaced by API calls for this purpose. It is still used as a transient database for authentication rate limiting and greylisting though, so it should not be removed from the compose file.
4. The test file is a very basic stub used to send an email, which I committed for the purpose of manual testing but is not ready yet for automated testing.
5. Indeed, the general idea is that once this is working, I will test the actual support of other databases (we might have some sqlite specific tricks in the data model still) and then add a configuration value for specifying the database location plus related documentation.

**Comment by kaiyou** [comment]
I believe issue 1. is now fixed. Could you spare some time with testing the latest commit?

**Comment by kaiyou** [comment]
Good news: some adjustments are needed to run Mailu against MySQL or Postgres, but those are quite minimal and I was just able to run a working Mailu instance backed by MySQL :)

**Comment by muhlemmer** [comment]
We'll take this up in the testing round tomorrow. @ionutfilip

**Comment by muhlemmer** [comment]
@kaiyou Did you see the conflict?

**Comment by kaiyou** [comment]
Yes, the conflict can be resolved at merge time and is pretty straightforward :)

**Comment by muhlemmer** [comment]
@kaiyou, I looked to resolving the merge conflict myself, but I'm not comfortable doing it. Especially for the `core/postfix/conf/master.cf` I have my doubts.

**Comment by kaiyou** [comment]
I'll resolve it tomorrow then :)

**Comment by kaiyou** [comment]
Merged, if you could have a quick look and see everything still runs fine, especially the tenacity part :)

File: core/postfix/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FROM alpine:3.8
 
 RUN apk add --no-cache postfix postfix-pcre rsyslog \
-    python3 py3-pip \
+ && python3 py3-pip \
  && pip3 install --upgrade pip
```
Comment (muhlemmer) [inline core/postfix/Dockerfile]: You need another backslash at the end of this line.

Comment (kaiyou) [reply core/postfix/Dockerfile]: God do I need more coffee.

**Comment by muhlemmer** [comment]
We might put an expresso machine on the "Donation list":
`ERROR: Service 'smtp' failed to build: The command '/bin/sh -c apk add --no-cache postfix postfix-pcre rsyslog  && python3 py3-pip  && pip3 install --upgrade pip  && pip3 install jinja2 podop tenacity' returned a non-zero code: 127`

File: core/postfix/Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@ FROM alpine:3.8
 
 RUN apk add --no-cache postfix postfix-pcre rsyslog \
  && python3 py3-pip \
```
Comment (muhlemmer) [inline core/postfix/Dockerfile]: Omit `&&` beginning of line

Comment (muhlemmer) [reply core/postfix/Dockerfile]: Basically, your first typo rectification was wrong itself. The Dockerfile was correct with the merge you did:
https://github.com/Mailu/Mailu/blob/00b5ae11db3f41d81c0b64b39bdddd99ce5a73a9/core/dovecot/Dockerfile

The Travis build for the merge commit failed only on running of rspamd, which is a known issue (#627).
https://travis-ci.org/Mailu/Mailu/builds/439491439#L1373
I've now re-triggered the build for that specific commit.

I think if you revert your last two commits in this branch (the typo's), we're good to go.

File: core/postfix/Dockerfile [hunk 3]
```diff
@@ -2,7 +2,7 @@ FROM alpine:3.8
...
- && pip3 install --upgrade pip \
+    pip3 install --upgrade pip \
```
Comment (muhlemmer) [inline core/postfix/Dockerfile]: @kaiyou  Wrong line, the build fails on line 3 /4. `apk add` needs to install `python3 and `py3-pip`. With the `&&` in front, you just try to run `python3` which is not installed.

**Comment by muhlemmer** [comment]
All back to normal again :p. It works fine now.

**Comment by muhlemmer** [comment]
:tada:  :balloon:  :birthday:

**Pull request merged**